lib/Bio/EnsEMBL/DataCheck/Checks/CheckFlatProteinTrees.pm

```diff
@@ -0,0 +1,122 @@
+=head1 LICENSE
+
+Copyright [2018-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::CheckFlatProteinTrees;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'CheckFlatProteinTrees',
+  DESCRIPTION    => 'Check protein tree integrity ensuring number of leaves with parent node at root < 3',
+  GROUPS         => ['compara', 'compara_protein_trees'],
+  DATACHECK_TYPE => 'critical',
+  DB_TYPES       => ['compara'],
+  TABLES         => ['gene_tree_node', 'gene_tree_root']
+};
+
+sub tests {
+  my ($self) = @_;
+  my $dba = $self->dba;
+  my $helper = $dba->dbc->sql_helper;
+
+  # Count and collect nodes with root_id != 0 and is not a seq_member
+  my $internal_node_sql = qq/
+    SELECT gtn.root_id, COUNT(*) AS internal_nodes 
+      FROM gene_tree_node gtn 
+    WHERE gtn.seq_member_id IS NULL 
+      AND gtn.node_id <> gtn.root_id 
+      AND gtn.root_id <> 0 
```

Comment:
0 used to the value for missing data (i.e. NULL) but we don't use it any more.
The column is now defined as a foreign key onto `gene_tree_node.node_id` and we have a check in the protein-tree pipeline to make sure it is never NULL.
In the Java HCs, compara/ForeignKeyGeneTreeTables.java test this:
```java
result &= checkForOrphans(con, "gene_tree_root", "root_id", "gene_tree_node", "node_id");
```
So I think this line can be removed